cmd/tools/releaser/internal/git/git.go

```diff
@@ -0,0 +1,91 @@
+package git
+
+import (
+	"bufio"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"go.uber.org/zap"
+)
+
+// Client implements Interface
+type Client struct {
+	logger *zap.Logger
+}
+
+func NewGitClient(logger *zap.Logger) *Client {
+	return &Client{logger: logger}
+}
+
+func (g *Client) GetCurrentBranch() (string, error) {
+	g.logger.Debug("Getting current git branch")
+	cmd := exec.Command("git", "rev-parse", "--abbrev-ref", "HEAD")
+	output, err := cmd.Output()
+	if err != nil {
+		return "", fmt.Errorf("get current branch: %w", err)
+	}
+	branch := strings.TrimSpace(string(output))
+	g.logger.Debug("Current branch", zap.String("branch", branch))
+	return branch, nil
+}
+
+func (g *Client) IsWorkingDirClean() (bool, error) {
+	g.logger.Debug("Checking if working directory is clean")
+	cmd := exec.Command("git", "diff-index", "--quiet", "HEAD", "--")
+	err := cmd.Run()
+	isClean := err == nil
+	g.logger.Debug("Working directory status", zap.Bool("clean", isClean))
+	return isClean, nil
+}
```

Comment:
not relevant for tags since we're only marking commits that already exist

Reply:
Make sense. Remove this check